src/System.Linq/tests/EnumerableTests.cs

```diff
@@ -172,6 +173,57 @@ public void Distinct()
                 "namespace",
             });
         }
+
+        [Fact]
+        public void VectorAddition()
+        {
+            int ARRAY_SIZE = 10000;
+            int[] array = RandomizeArray(ARRAY_SIZE);
+
+            int sum = 0;
+            int simdSum = 0;
+            // Sum those numbers using the tradition LINQ
+            Stopwatch sw = Stopwatch.StartNew();
+
+            for (int i = 0; i < 1000; i++)
+            {
+                // hard-coding in the "traditional" addition method from IEnumerable<int>.Sum()
+                foreach (int num in array)
```

Comment:
This is not quite the same as the Linq method. This would use `System.Array+SZArrayEnumerator` whereas Linq would be using `System.SZArrayHelper+SZGenericArrayEnumerator<int>`. While the results are likely not very different, you're measuring the cost of something similar to the Linq approach here, not the linq approach.